src/libs/logging/src/lib.rs

```diff
@@ -3,26 +3,67 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+#[macro_use]
+extern crate lazy_static;
+use arc_swap::ArcSwap;
 use slog::{o, record_static, BorrowedKV, Drain, Key, OwnedKV, OwnedKVList, Record, KV};
+
 use std::collections::HashMap;
 use std::io;
 use std::io::Write;
 use std::process;
 use std::result;
-use std::sync::Mutex;
+use std::sync::{Arc, Mutex};
 
 mod file_rotate;
 mod log_writer;
 
 pub use file_rotate::FileRotator;
 pub use log_writer::LogWriter;
 
+lazy_static! {
+    pub static ref AGENT_LOGGER: ArcSwap<slog::Logger> =
+        ArcSwap::from(Arc::new(slog::Logger::root(slog::Discard, o!())));
+    pub static ref HYPERVISOR_LOGGER: ArcSwap<slog::Logger> =
+        ArcSwap::from(Arc::new(slog::Logger::root(slog::Discard, o!())));
+    pub static ref RESOURCE_LOGGER: ArcSwap<slog::Logger> =
+        ArcSwap::from(Arc::new(slog::Logger::root(slog::Discard, o!())));
+    pub static ref RUNTIMES_LOGGER: ArcSwap<slog::Logger> =
+        ArcSwap::from(Arc::new(slog::Logger::root(slog::Discard, o!())));
+    pub static ref VIRT_CONTAINER_LOGGER: ArcSwap<slog::Logger> =
+        ArcSwap::from(Arc::new(slog::Logger::root(slog::Discard, o!())));
+    pub static ref SERVICE_LOGGER: ArcSwap<slog::Logger> =
+        ArcSwap::from(Arc::new(slog::Logger::root(slog::Discard, o!())));
+    pub static ref SHIM_LOGGER: ArcSwap<slog::Logger> =
+        ArcSwap::from(Arc::new(slog::Logger::root(slog::Discard, o!())));
+    
+    pub static ref VMM_LOGGER: ArcSwap<slog::Logger> =
```

Comment:
I wonder why there are  VMM_LOGGER, HYPERVISOR_LOGGER and VMM_DRAGONBALL_LOGGER?

How about writing some comments?